bundles/org.openhab.binding.froniuswattpilot/src/main/java/org/openhab/binding/froniuswattpilot/internal/FroniusWattpilotMDNSDiscoveryServiceParticipant.java

```diff
@@ -0,0 +1,110 @@
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.config.discovery.mdns.MDNSDiscoveryParticipant;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+
+/**
+ * The {@link FroniusWattpilotMDNSDiscoveryServiceParticipant} is responsible for discovering new and removed Wattpilot
+ * wallboxes through mDNS. It uses the central
+ * {@link org.openhab.core.config.discovery.mdns.internal.MDNSDiscoveryService}.
+ *
+ * @author Florian Hotze - Initial contribution
+ */
+@Component(service = MDNSDiscoveryParticipant.class, configurationPid = "discovery.froniuswattpilot")
+@NonNullByDefault
+public class FroniusWattpilotMDNSDiscoveryServiceParticipant implements MDNSDiscoveryParticipant {
+
+    private static final String SERVICE_TYPE = "_Fronius-SE-Wattpilot._tcp.local.";
+
+    private boolean isAutoDiscoveryEnabled = true;
+
+    @Activate
+    protected void activate(ComponentContext componentContext) {
+        activateOrModifyService(componentContext);
+    }
+
+    @Modified
+    protected void modified(ComponentContext componentContext) {
+        activateOrModifyService(componentContext);
+    }
+
+    private void activateOrModifyService(ComponentContext componentContext) {
+        Dictionary<String, @Nullable Object> properties = componentContext.getProperties();
+        String autoDiscoveryPropertyValue = (String) properties
+                .get(DiscoveryService.CONFIG_PROPERTY_BACKGROUND_DISCOVERY);
+        if (autoDiscoveryPropertyValue != null && !autoDiscoveryPropertyValue.isBlank()) {
+            isAutoDiscoveryEnabled = Boolean.valueOf(autoDiscoveryPropertyValue);
+        }
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    public String getServiceType() {
+        return SERVICE_TYPE;
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(ServiceInfo service) {
+        if (isAutoDiscoveryEnabled) {
+            ThingUID uid = getThingUID(service);
+            if (uid != null) {
+                String[] addresses = service.getHostAddresses();
+                if (addresses == null || addresses.length == 0) {
+                    return null;
+                }
+                String ip = addresses[0];
+                return DiscoveryResultBuilder.create(uid) //
+                        .withProperties(Map.of( //
+                                "hostname", ip)) //
+                        .withLabel(service.getName()) //
+                        .withRepresentationProperty("hostname") //
```

Comment:
Woudl be better to extract the configuration key to a binding level string constant